### PR TITLE
Add task to generate the repo to repo graph

### PR DESCRIPTION
**Pull request by natemcmaster**
Porting https://github.com/aspnet/Universe/pull/696 to dev, minus the part that generates .repoproj files.

The graph now:

image

### CODE REVIEW

**Comment by natemcmaster** [comment]
Ooh, VS can do analysis. 
Purple = leaf nodes,
Pink = "hubs" (i.e. the most referenced repos)

image

File: build/tasks/CodeGen/GenerateSubmoduleGraph.cs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+            for (var i = 0; i < solutions.Count; i++)
+            {
+                var sln = repoGraph[i] = solutions[i];
+
+                foreach (var proj in sln.Projects)
+                {
+                    if (!proj.IsPackable
+                        || proj.FullPath.Contains("samples")
+                        || proj.FullPath.Contains("tools/VS.Web.CG.Design"))
```
Comment (pranavkm) [inline build/tasks/CodeGen/GenerateSubmoduleGraph.cs]: Out of curiosity, what's special about this?

Comment (natemcmaster) [reply build/tasks/CodeGen/GenerateSubmoduleGraph.cs]: There are multiple csproj files in this folder that produce the same package ID, just for different platforms (x64/arm/x86). We should probably just fix scaffolding's code.

[the remark this answers is not in the document]
Comment (natemcmaster) [reply build/tasks/CodeGen/GenerateSubmoduleGraph.cs]: https://github.com/aspnet/Scaffolding/pull/726 - this fixes it

**Pull request merged**